src/herder/HerderImpl.cpp

```diff
@@ -1457,8 +1457,8 @@ HerderImpl::triggerNextLedger(uint32_t ledgerSeqToTrigger,
     // see if we need to include some upgrades
     std::vector<LedgerUpgrade> upgrades;
     {
-        LedgerTxn ltx(mApp.getLedgerTxnRoot());
-        upgrades = mUpgrades.createUpgradesFor(lcl.header, ltx);
+        LedgerSnapshot ls(mApp);
+        upgrades = mUpgrades.createUpgradesFor(lcl.header, ls);
```

Comment:
nit: Is the scope still necessary here?

Reply:
i think it wasn't strictly needed to begin with, but I'd prefer to keep it so we don't unnecessarily extend lifetime of LedgerTxns